src/context/auth-reducer.js

```
@@ -6,6 +6,12 @@ export function authReducer(state, { type, payload }) {
         token: payload.encodedToken,
         userName: payload.foundUser?.firstName,
       };
+    case "LOG_OUT":
+      return {
+        ...state,
+        token: payload.token,
+        userName: payload.userName,
+      };
     default:
       throw new Error("Unhandled type in auth reducer");
```

great work here like the way you throw error instead of returning state value